pkg/runtime/internal/envdef/environment.go

```diff
@@ -125,6 +126,24 @@ func NewEnvironmentDefinition(fp string) (*EnvironmentDefinition, error) {
 	if err != nil {
 		return nil, errs.Wrap(err, "could not unmarshal environment definition file: %s", fp)
 	}
+
+	if ignores := os.Getenv(constants.IgnoreEnvEnvVarName); ignores != "" {
+		ignore := make(map[string]bool)
+		for _, name := range strings.Split(ignores, ",") {
+			ignore[name] = true
+		}
+
+		// Remove any environment variables to ignore.
+		for i := 0; i < len(ed.Env); {
+			if _, exists := ignore[ed.Env[i].Name]; !exists {
+				i++
```

Comment:
If the first entry matches it seems like you would have an infinite loop here, because `i` will forever remain `0`.